### PR TITLE
Added HEADER_CHECKS configuration option

### PR DESCRIPTION
**Pull request by malware-watch**
## Description

I added the HEADER_CHECKS [1] option with a sample configuration.

Each message header line is compared against a pre-configured list of patterns. When a match is found the corresponding action is executed.

## Type of change

- [ ] New feature (non-breaking change which adds functionality)

## How Has This Been Tested?

I have tested this version of the docker image in a flask application and checked if the headers where ignored or replaced.

[1]: http://www.postfix.org/header_checks.5.html

### CODE REVIEW

File: header_checks [hunk 1]
```diff
@@ -0,0 +1,11 @@
+# Sample For Dropping Headers: 	
+#/^Header: IfContains/ 	      IGNORE
+/^Received:.*with ESMTPSA/    IGNORE
+/^Received:.*with ESMTPS/     IGNORE
+/^Received:.*with SMTP/       IGNORE
+/^Received:/ 	              IGNORE
```
Comment (bokysan) [inline header_checks]: Can you just make sure that the lines are aligned?

File: README.md [hunk 2]
```diff
@@ -138,6 +138,10 @@ Example:
 docker run --rm --name postfix -e "ALLOWED_SENDER_DOMAINS=example.com example.org" -e "MASQUERADED_DOMAINS=example.com" -p 1587:587 boky/postfix
 ```
 
+### `HEADER_CHECKS`
+
+Each message header line is compared against a pre-configured list of patterns. When a match is found the corresponding action is executed. Set to a non-empty string (usually "1" or "yes") enable.
+
```
Comment (bokysan) [inline README.md]: We'll need additional description here -- e.g. instructions on how to add the file. This is not apparent from documentation.

**Comment by malware-watch** [comment]
Thank you for your review.
I tried to add a better description and corrected the formatting.

Can you please check again?

**Comment by bokysan** [comment]
Thank you, I'll merge this into the main branch but I'll most likely move the `header_checks` file to another location to make it easier to override it.

**Comment by bokysan** [comment]
Hi,

after consideration I've renamed your variable to `SMTP_HEADER_CHECKS` to better reflect the variable being set.

**Comment by malware-watch** [comment]
Makes sense.
Thanks for merging!

**Pull request merged**